HTTP 500 errors after upgrading to Flask 2.3.0

Download requests fail with HTTP 500 responses after upgrading to Flask >= 2.3.0. The traceback is:

```
[ERROR] AttributeError: 'Flask' object has no attribute 'before_first_request'
Traceback (most recent call last):
  File "/var/lang/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/var/task/door/__init__.py", line 5, in <module>
    from door import routes  # noqa Has to be at end of file or will cause circular import
  File "/var/task/door/routes.py", line 28, in <module>
    @app.before_first_request
```

`app.before_first_request` was deprecated in [Flask 2.2.0](https://github.com/pallets/flask/blob/main/CHANGES.rst#version-220) and removed in [Flask 2.3.0](https://github.com/pallets/flask/blob/main/CHANGES.rst#version-230).

Our app uses `before_first_request` at https://github.com/asfadmin/grfn-distribution/blob/test/door/src/door/routes.py#L28 to fetch a private key from secrets manager once when the app is initialized. We'll need to refactor where that initialization is done, or pursue https://github.com/asfadmin/grfn-distribution/pull/204 which eliminates the need for the private key.

--- a/door/src/door/lambda_handler.py
+++ b/door/src/door/lambda_handler.py
@@ -1,9 +1,24 @@
+import json
+import os
+
+import boto3
 import serverless_wsgi
 
 from door import app
 
 
 serverless_wsgi.TEXT_MIME_TYPES.append('application/problem+json')
+
+
+def get_secret(secret_name):
+    sm = boto3.client('secretsmanager', os.environ['AWS_REGION'])
+    response = sm.get_secret_value(SecretId=secret_name)
+    secret = json.loads(response['SecretString'])
+    return secret
+
+
+private_key = get_secret(os.environ['PRIVATE_KEY_SECRET_NAME'])['private_key']
+os.environ['CLOUDFRONT_PRIVATE_KEY'] = str(private_key)
 
 
 def handler(event, context):

--- a/door/src/door/routes.py
+++ b/door/src/door/routes.py
@@ -1,4 +1,3 @@
-import json
 import os
 from datetime import datetime, timedelta, timezone
 from urllib.parse import quote_plus
@@ -23,12 +22,6 @@ def decode_token(token):
         return payload
     except (jwt.ExpiredSignatureError, jwt.DecodeError):
         return None
-
-
-@app.before_first_request
-def init_app():
-    private_key = get_secret(os.environ['PRIVATE_KEY_SECRET_NAME'])['private_key']
-    os.environ['CLOUDFRONT_PRIVATE_KEY'] = str(private_key)
 
 
 @app.before_request
@@ -71,10 +64,3 @@ def get_signed_url(object_key, user_id, private_key):
     cf_signer = CloudFrontSigner(os.environ['CLOUDFRONT_KEY_PAIR_ID'], rsa_signer)
     signed_url = cf_signer.generate_presigned_url(base_url, date_less_than=expiration_datetime)
     return signed_url
-
-
-def get_secret(secret_name):
-    sm = boto3.client('secretsmanager', os.environ['AWS_REGION'])
-    response = sm.get_secret_value(SecretId=secret_name)
-    secret = json.loads(response['SecretString'])
-    return secret

